dipy/data/__init__.py

```diff
@@ -313,3 +314,51 @@ def matlab_life_results():
     matlab_rmse = np.load(pjoin(DATA_DIR, 'life_matlab_rmse.npy'))
     matlab_weights = np.load(pjoin(DATA_DIR, 'life_matlab_weights.npy'))
     return matlab_rmse, matlab_weights
+
+
+def load_sdp_constraints(id, order=None):
+    """Import semidefinite programming constraint matrices for different models,
+    generated as described for example in [1]_.
+
+    Parameters
+    ----------
+    id : string
+        A string identifying the model that is to be constrained.
+    order : unsigned int
+        A non-negative integer that represent the order or instance of the
+        model.
+        Default: None.
+
+    Returns
+    -------
+    sdp_constraints : array
+        Constraint matrices
+
+    Notes
+    -----
+    The constraints will be loaded from a file named 'id_constraint_order.npz'.
+
+    References
+    ----------
+    .. [1] Dela Haije et al. "Enforcing necessary non-negativity constraints
+           for common diffusion MRI models using sum of squares programming".
+           NeuroImage 209, 2020, 116405.
+
+    """
+
+    file = id + '_constraint'
+    if order is not None:
+        file += '_' + str(order)
+    file += '.npz'
+    path = pjoin(DATA_DIR, file)
+
+    if not exists(path):
+        raise ValueError("Constraints file '" + file + "' not found.")
+
+    try:
+        array = load_npz(path)
+        n, x = array.shape
+        sdp_constraints = [array[i*x:(i+1)*x] for i in range(n//x)]
```

Comment:
Thanks for your comments @StongeEtienne!

What do you mean with "0.0 * v"? 

CVXPY does make use of sparsity, and the code runs faster with sparse matrices than with dense matrices. Because `reshape` makes the matrix dense, and because there are no 3D sparse matrix representations, I had to use this list reconstruction approach. Do you have ideas on how to improve this part while keeping the sparse representation?

Reply:
Sum optimizer, avoid variables multiplied by 0,
I compared with and without dense matrix and the procession (optimisation) time was about the same for order 6.
I will think of something, but the matrix computation is not the "bottleneck" so I dont really mind.